wstring_convert() deprecation warning

Found when tackling issue: https://github.com/mantidproject/mantid/issues/33801

**Describe the bug**
As mentioned [here](https://github.com/mantidproject/mantid/issues/33801#issuecomment-1150089236), there is a warning when building with 
`-DUSE_PRECOMPILED_HEADERS=OFF`.

<details>
    <summary>UnitLabel.cppL28: wstring_convert() deprecation warning</summary>

    C:\Users\mlc47243\Documents\mantid-development\mantid-conda\mantid\Framework\Kernel\src\UnitLabel.cpp(28,29): warning C4996: 'std::codecvt_utf8_utf16<wchar_t,1114111,(std::codecvt_mode)0>': warning STL4017: std::wbuffer_convert, std::wstring_convert, and the <codecvt> header (containing std::codecvt_mode, std::codecvt_utf8, std::codecvt_utf16, and std::codecvt_utf8_utf16) are deprecated in C++17. (The std::codecvt class template is NOT deprecated.) The C++ Standard doesn't provide equivalent non-deprecated functionality; consider using MultiByteToWideChar() and WideCharToMultiByte() from <Windows.h> instead. You can define _SILENCE_CXX17_CODECVT_HEADER_DEPRECATION_WARNING or _SILENCE_ALL_CXX17_DEPRECATION_WARNINGS to acknowledge that you have received this warning. [C:\Users\mlc47243\Documents\mantid-development\mantid-conda\build\Framework\Kernel\Kernel.vcxproj]
    C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.27.29110\include\codecvt(428,5): message : see declaration of 'std::codecvt_utf8_utf16' (compiling source file C:\Users\mlc47243\Documents\mantid-development\mantid-conda\mantid\Framework\Kernel\src\UnitLabel.cpp) [C:\Users\mlc47243\Documents\mantid-development\mantid-conda\build\Framework\Kernel\Kernel.vcxproj]
    C:\Users\mlc47243\Documents\mantid-development\mantid-conda\mantid\Framework\Kernel\src\UnitLabel.cpp(28,8): warning C4996: 'std::wstring_convert<std::codecvt_utf8_utf16<wchar_t,1114111,(std::codecvt_mode)0>,wchar_t,std::allocator<wchar_t>,std::allocator<char>>': warning STL4017: std::wbuffer_convert, std::wstring_convert, and the <codecvt> header (containing std::codecvt_mode, std::codecvt_utf8, std::codecvt_utf16, and std::codecvt_utf8_utf16) are deprecated in C++17. (The std::codecvt class template is NOT deprecated.) The C++ Standard doesn't provide equivalent non-deprecated functionality; consider using MultiByteToWideChar() and WideCharToMultiByte() from <Windows.h> instead. You can define _SILENCE_CXX17_CODECVT_HEADER_DEPRECATION_WARNING or _SILENCE_ALL_CXX17_DEPRECATION_WARNINGS to acknowledge that you have received this warning. [C:\Users\mlc47243\Documents\mantid-development\mantid-conda\build\Framework\Kernel\Kernel.vcxproj]
    C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.27.29110\include\xlocbuf(301,55): message : see declaration of 'std::wstring_convert' (compiling source file C:\Users\mlc47243\Documents\mantid-development\mantid-conda\mantid\Framework\Kernel\src\UnitLabel.cpp) [C:\Users\mlc47243\Documents\mantid-development\mantid-conda\build\Framework\Kernel\Kernel.vcxproj]
    C:\Users\mlc47243\Documents\mantid-development\mantid-conda\mantid\Framework\Kernel\src\UnitLabel.cpp(28,58): warning C4996: 'std::wstring_convert<std::codecvt_utf8_utf16<wchar_t,1114111,(std::codecvt_mode)0>,wchar_t,std::allocator<wchar_t>,std::allocator<char>>::wstring_convert': warning STL4017: std::wbuffer_convert, std::wstring_convert, and the <codecvt> header (containing std::codecvt_mode, std::codecvt_utf8, std::codecvt_utf16, and std::codecvt_utf8_utf16) are deprecated in C++17. (The std::codecvt class template is NOT deprecated.) The C++ Standard doesn't provide equivalent non-deprecated functionality; consider using MultiByteToWideChar() and WideCharToMultiByte() from <Windows.h> instead. You can define _SILENCE_CXX17_CODECVT_HEADER_DEPRECATION_WARNING or _SILENCE_ALL_CXX17_DEPRECATION_WARNINGS to acknowledge that you have received this warning. [C:\Users\mlc47243\Documents\mantid-development\mantid-conda\build\Framework\Kernel\Kernel.vcxproj]
    C:\Users\mlc47243\Documents\mantid-development\mantid-conda\mantid\Framework\Kernel\src\UnitLabel.cpp(29,21): warning C4996: 'std::wstring_convert<std::codecvt_utf8_utf16<wchar_t,1114111,(std::codecvt_mode)0>,wchar_t,std::allocator<wchar_t>,std::allocator<char>>::from_bytes': warning STL4017: std::wbuffer_convert, std::wstring_convert, and the <codecvt> header (containing std::codecvt_mode, std::codecvt_utf8, std::codecvt_utf16, and std::codecvt_utf8_utf16) are deprecated in C++17. (The std::codecvt class template is NOT deprecated.) The C++ Standard doesn't provide equivalent non-deprecated functionality; consider using MultiByteToWideChar() and WideCharToMultiByte() from <Windows.h> instead. You can define _SILENCE_CXX17_CODECVT_HEADER_DEPRECATION_WARNING or _SILENCE_ALL_CXX17_DEPRECATION_WARNINGS to acknowledge that you have received this warning. [C:\Users\mlc47243\Documents\mantid-development\mantid-conda\build\Framework\Kernel\Kernel.vcxproj]  

</details>

This flag isn't too important but the warning may be silenced somewhere else in the codebase. More thought is needed as to whether we need this wstring_convert and sothis deserves a separate issue

**To Reproduce**
run ``cmake -DUSE_PRECOMPILED_HEADERS=OFF ..``

**Additional context**
This funciton is deprecated in C++17 but still exists (deprecated) in C++20 so we could make no code change and silence the warning
We should review the code that is currrently using it to check that it is even needed in the first place.

--- a/Framework/Algorithms/CMakeLists.txt
+++ b/Framework/Algorithms/CMakeLists.txt
@@ -1045,7 +1045,9 @@ endif()
 enable_precompiled_headers(inc/MantidAlgorithms/PrecompiledHeader.h SRC_FILES)
 # Disable optimization for certain Segfault as the key lines that cause the segfault can be optimized away but this
 # algorithm is only ever intended to test the crashing behaviour of the framework
-set_source_files_properties(src/Segfault.cpp PROPERTIES COMPILE_FLAGS -O0)
+if(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
+  set_source_files_properties(src/Segfault.cpp PROPERTIES COMPILE_FLAGS -O0)
+endif()
 
 # Add the target for this directory
 add_library(Algorithms ${SRC_FILES} ${C_SRC_FILES} ${INC_FILES})

--- a/buildconfig/CMake/WindowsSetup.cmake
+++ b/buildconfig/CMake/WindowsSetup.cmake
@@ -17,6 +17,9 @@ add_definitions(-DBOOST_CONFIG_SUPPRESS_OUTDATED_MESSAGE)
 add_definitions(-D_SCL_SECURE_NO_WARNINGS -D_CRT_SECURE_NO_WARNINGS)
 # Prevent deprecation errors from std::tr1 in googletest until it is fixed upstream. In MSVC 2017 and later
 add_definitions(-D_SILENCE_TR1_NAMESPACE_DEPRECATION_WARNING)
+# Suppress codecvt deprecation warning when building with Ninja. Using codecvt is still the recommended solution for
+# converting from std::string to std::wstring. See https://stackoverflow.com/a/18597384
+add_definitions(-D_SILENCE_CXX17_CODECVT_HEADER_DEPRECATION_WARNING)
 # Suppress warnings about std::iterator as a base. TBB emits this warning and it is not yet fixed.
 add_definitions(-D_SILENCE_CXX17_ITERATOR_BASE_CLASS_DEPRECATION_WARNING)
 add_definitions(-D_SILENCE_CXX17_SHARED_PTR_UNIQUE_DEPRECATION_WARNING)

--- a/qt/scientific_interfaces/Indirect/CMakeLists.txt
+++ b/qt/scientific_interfaces/Indirect/CMakeLists.txt
@@ -170,11 +170,9 @@ set(MOC_FILES
     IndirectFitPropertyBrowser.h
     InelasticDataManipulation.h
     InelasticDataManipulationElwinTab.h
-    InelasticDataManipulationElwinTabModel.h
     InelasticDataManipulationElwinTabView.h
     InelasticDataManipulationIqtTab.h
     InelasticDataManipulationIqtTabView.h
-    InelasticDataManipulationIqtTabModel.h
     InelasticDataManipulationMomentsTab.h
     InelasticDataManipulationMomentsTabView.h
     InelasticDataManipulationSqwTab.h
@@ -207,6 +205,8 @@ set(INC_FILES
     IndirectFunctionBrowser/ConvFunctionModel.h
     IndirectPlotOptionsModel.h
     IndirectSettingsHelper.h
+    InelasticDataManipulationElwinTabModel.h
+    InelasticDataManipulationIqtTabModel.h
     InelasticDataManipulationMomentsTabModel.h
     InelasticDataManipulationSqwTabModel.h
     InelasticDataManipulationSymmetriseTabModel.h

